packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -1941,7 +1941,10 @@ class TestWidgetInspectorService extends Object with WidgetInspectorService {
       // Verify that taking a screenshot did not change the number of children
       // of the layer.
       expect(getChildLayerCount(layer), equals(expectedChildLayerCount));
-
+      // Remove test name to avoid duplicate test exception.
+      // Check if it is present first in cases of skipped test.
+      if (goldenTestNames.contains('inspector.repaint_boundary_margin.png'))
+        goldenTestNames.remove('inspector.repaint_boundary_margin.png');
```

Comment:
Would it be a nicer API if instead of modifying the set directly, you could set a parameter during the call to matchesGoldenFile indicating that re-using this filename is ok?

```dart
  matchesGoldenFile('inspector.repaint_boundary_margin.png', filenameReused: true),
```

Probably needs a better name, but would avoid introducing another global.

Reply:
That is much cleaner. 🧼